server: add tests for combined stmts api source tables

xinhaoz opened the pull request:
This patch creates the sqlstatstestutils pkg in
sqlstats/persistedsqlstats.  It moves the function
`GetRandomizedCollectedStatementStatisticsForTest` which was
previously in sqlstatsutil. This pkg is introduced in
preparation for adding more testing utilities to mock sql
stats tables without creating cyclic dependencies with
sqlstatsutil.

Epic: none

Release note: None

### 2: [server: add tests for combined stmts api](https://github.com/cockroachdb/cockroach/commit/1bf95c1d6dd1a680b9944a730020eee26c5bf580)

Add testing to verify that the combined stmts api uses the
correct source tables given different table states.

Epic: none

xinhaoz committed:
sqlstats: create sqlstatstestutils pkg

This patch creates the sqlstatstestutils pkg in
sqlstats/persistedsqlstats.  It moves the function
`GetRandomizedCollectedStatementStatisticsForTest` which was
previously in sqlstatsutil. This pkg is introduced in
preparation for adding more testing utilities to mock sql
stats tables without creating cyclic dependencies with
sqlstatsutil.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/112106)
<!-- Reviewable:end -->

xinhaoz committed:
server: add tests for combined stmts api source tables

Add testing to verify that the combined stmts api uses the
correct source tables given different table states.

Epic: none

Release note: None

maryliag reviewed (commented):
> Reviewed 7 of 7 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112106)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier and @xinhaoz)

___
*[`pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/sqlstatstestutil/testutils.go` line 365 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/112106#-NgPc35X5hBXTEip9FSB:-NgPc35X5hBXTEip9FSC:b-v24rff) ([raw file](https://github.com/cockroachdb/cockroach/blob/d005e98d535d5a93a68b0ac5b3f0e93f3fd36295/pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/sqlstatstestutil/testutils.go#L365)):*
> ```Go
> 		statistics,             // statistics
> 		plan,                   // plan
> 		// TODO allow these values to be mocked. No need ffor them for now.
> ```

nit: typo ffor


<!-- Sent from Reviewable.io -->

maryliag reviewed (approved):
just one nit, otherwise <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 5 of 14 files at r3, 5 of 5 files at r4, 5 of 9 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112106#-:-Ni00Sv02M8IDsj9btNo:bcy4bus)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @abarganier and @xinhaoz)

___
*[`pkg/sql/sqlstats/persistedsqlstats/sqlstatstestutil/testutils.go` line 1 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/112106#-Ni0075j67Ca4rUw7ubP:-Ni0075j67Ca4rUw7ubQ:bxqvv2x) ([raw file](https://github.com/cockroachdb/cockroach/blob/27d65b6383d1848f52e8e51e2dba4d7b413a02ad/pkg/sql/sqlstats/persistedsqlstats/sqlstatstestutil/testutils.go#L1)):*
> ```Go
> // Copyright 2021 The Cockroach Authors.
> ```

nit: 2023


<!-- Sent from Reviewable.io -->

xinhaoz commented:
tftr!
bors r=maryliag
